**Import Libraries**

In [1]:
import pandas as pd
import numpy as np
from sklearn.model_selection import train_test_split
from sklearn.ensemble import AdaBoostClassifier
import warnings
warnings.filterwarnings("ignore")

**Reading  And Describing The Dataset** 

In [2]:
from google.colab import files
uploaded = files.upload()

Saving Iris.csv to Iris.csv


In [3]:
# Reading the dataset from the csv file
# separator is a vertical line, as seen in the dataset
data = pd.read_csv("Iris.csv")

# Printing the shape of the dataset
print(data.shape)

(150, 6)


In [4]:
data.head()


Id  SepalLengthCm  SepalWidthCm  PetalLengthCm  PetalWidthCm      Species
0   1            5.1           3.5            1.4           0.2  Iris-setosa
1   2            4.9           3.0            1.4           0.2  Iris-setosa
2   3            4.7           3.2            1.3           0.2  Iris-setosa
3   4            4.6           3.1            1.5           0.2  Iris-setosa
4   5            5.0           3.6            1.4           0.2  Iris-setosa

**Dropping Irrevelant Column and Separating Target Variable**

In [5]:
data = data.drop('Id',axis=1)
X = data.iloc[:,:-1]
y = data.iloc[:,-1]
print("Shape of X is %s and shape of y is %s"%(X.shape,y.shape))


Shape of X is (150, 4) and shape of y is (150,)


**Unique values in our Target Variable**

In [6]:
total_classes = y.nunique()
print("Number of unique species in dataset are: ",total_classes)


Number of unique species in dataset are:  3


In [7]:
distribution = y.value_counts()
print(distribution)


Iris-setosa        50
Iris-versicolor    50
Iris-virginica     50
Name: Species, dtype: int64


**Splitting The Dataset**

In [8]:
X_train, X_val, Y_train, Y_val = train_test_split(
	X, y, test_size=0.25, random_state=28)


**Applying AdaBoost**

In [9]:
# Creating adaboost classifier model
adb = AdaBoostClassifier()
adb_model = adb.fit(X_train,Y_train)


**Accuracy of the AdaBoost Model**

In [10]:
print("The accuracy of the model on validation set is", adb_model.score(X_val,Y_val))


The accuracy of the model on validation set is 0.9210526315789473


The model has an accuracy of 92.11% (approximately) on the validation set which is quite good with no hyperparameter tuning and feature engineering.